packages/react-ui-core/src/Ratings/RatingBar.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { randomId } from '@rentpath/react-ui-utils'
+import Star from './SampleSvgs/Star'
+
+export default class RatingBar extends Component {
+  static propTypes = {
+    theme: PropTypes.object,
+    ratingScale: PropTypes.number,
+    rating: PropTypes.number,
+    total: PropTypes.string,
+    RatingItem: PropTypes.node,
+  }
+  static defaultProps = {
+    theme: {},
+    RatingItem: Star,
+  }
+
+  get ratingItems() {
+    const {
+      ratingScale,
+      rating,
+      RatingItem,
+      ...props
+    } = this.props
+
+    return [...Array(ratingScale).keys()].map(index => (
+      <RatingItem
+        key={randomId('rating')}
+        id={randomId('rating-item')}
```

Comment:
The purpose of the `key` prop is to help `React` determine whether the component needs to be re-rendered/inserted into the DOM.  By generating a random key every time this component renders you are blocking a potential performance enhancement.  I think it would probably be suitable to just use the `index` value here.

Also, instead of generating a random `id` here every time this renders, lets move that responsibility into the `Star` component, and have it generate it either in the constructor, or create it _lazily_ like [here](https://github.com/rentpath/react-ui/blob/master/packages/react-ui-core/src/Form/Field.js#L118).

Reply:
That sort of makes sense, but the `id` is really only used internally by the `Star` component to link the svg nodes, so imo this parent component has no need to know about it.